src/libs/deploy.js

```diff
@@ -178,11 +180,131 @@ export const deployAppleStore = async (signedApp, workspace = '/tmp/') => {
+ * @param {String} awOrgID The airwatch organization group ID to distribute the app
+ * @param {String} awFileName The name of the app to appear in airwatch
  * @param {string} [workspace='/tmp/'] The workspace to use
  * @returns The status of the deployment
  */
-// eslint-disable-next-line no-unused-vars
-export const deployAirWatch = async (signedApp, workspace = '/tmp/') => {
-  // TODO
+// eslint-disable-next-line import/prefer-default-export
+export const deployAirWatch = async (signedApp, platform, awOrgID, awFileName, workspace = '/tmp/') => {
+  /*
+  Values for airwatch api v8_1:
+    DeviceType: android -> '5'; Apple -> '2'
+    ModelId: android -> 5; iPhone -> 1; iPad -> 2
+    ApplicationName: android -> apk; Apple -> ipa
+  */
+
+  let deviceType = '';
+  let applicationName = '';
+  let modelId = 0;
+
+  switch (platform) {
+    case 'ios':
+    {
+      applicationName = awFileName + '.ipa';
+      deviceType = '2';
+      modelId = 1;
+      break;
+    }
+    case 'android':
+    {
+      applicationName = awFileName + '.apk';
+      deviceType = '5';
+      modelId = 5;
+      break;
+    }
+    default:
+      throw new Error('Unsupported application type for airWatch deployment');
+  }
+
+  // The constant url for api:
+  const awHost = 'https://mobileconsole.gov.bc.ca';
+  const awUploadAPI = '/API/mam/blobs/uploadblob';
+  const awInstallAPI = '/API/mam/apps/internal/begininstall';
+
+  // Update the user account to a device-account:
+  const awUsernameF = await exec('security find-generic-password -w -s awUsername');
+  const awPasswordF = await exec('security find-generic-password -w -s awPassword');
+  const awTenantCodeF = await exec('security find-generic-password -w -s awCode');
+
+  // Extract value from stdout:
+  const awUsername = awUsernameF.stdout.trim().split('\n')[0];
+  const awPassword = awPasswordF.stdout.trim().split('\n')[0];
+  const awTenantCode = awTenantCodeF.stdout.trim().split('\n')[0];
+  
+  console.log(awUsername);
+  console.log(awPassword);
+  console.log(awTenantCode);
+
+  // Get app binary:
+  const signedAppPath = await fetchFileFromStorage(signedApp, workspace);
+  const appBinary = require('fs').readFileSync(signedAppPath);
+
+  console.log('Start to deploy to airwatch..');
+
+  // Step 1: Upload app as blob
+  const uploadOptions = {
+    headers: {
+      'Content-Type': 'application/octet-stream',
+      'aw-tenant-code': awTenantCode,
+      Accept: 'application/json',
+    },
+    auth: {
+      user: awUsername,
+      password: awPassword,
+    },
+    method: 'POST',
+    uri: url.resolve(awHost, awUploadAPI),
+    encoding: null,
+    body: appBinary,
+    qs: {
+      filename: signedApp,
+      organizationgroupid: awOrgID,
+    },
+  };
+
+  try {
+    const awUploadRes = await request(uploadOptions);
+
+    // get the blob id:
+    const blobID = JSON.parse(awUploadRes.toString()).Value;
+    console.log('The blob id is ' + blobID);
```

Comment:
Use `logger.info` or remove. Also use string templates where possible \`The blob ID is ${blobID}\`